Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -223,18 +223,38 @@ - (void)sendRequest:(NSURLRequest *)request onSuccess:(void (^)(NSDictionary *))
                                                                    NSURLResponse *response,
                                                                    NSError *error)
     {
-        if ([data length] > 0 && error == nil) {
-            error = nil;
-            id object = [NSJSONSerialization
-                         JSONObjectWithData:data
-                         options:0
-                         error:&error];
-            if(error) {
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+        NSInteger responseCode = httpResponse.statusCode;
+        
+        NSError *jsonError = nil;
+        id json = [NSJSONSerialization
+                     JSONObjectWithData:data
+                     options:0
+                     error:&jsonError];
+
+        if (responseCode == 401) {
+            if (onFailure != nil) onFailure(@"Invalid API Key", data);
+        } else if (responseCode >= 400) {
+            if (onFailure != nil) {
+                NSString *errorMessage = @"Invalid Request";
+
+                // Extract error message from the response JSON
+                if ([json isKindOfClass:[NSDictionary class]] && [json objectForKey:@"msg"] &&
```

Comment:
is it possible for `json` to be `nil`, and if it is, does `[nil isKindOfClass]` cause a crash?

Reply:
In Objective C, sending messages to nil does nothing and returns nil.